implementation.md

```diff
@@ -1554,3 +1554,268 @@ In the `Video` and `Audio` objects, sellers must only include one form of floor
+
+Different iterations of possible combinations are listed in the following table. Additional json examples for how to use <code>atype</code> alongside match method (<code>mm</code>) signaling can be found in the next section.
+
+<table>
+  <tr>
+    <td><strong>atype Value</strong></td>
+    <td><strong>atype Definition</strong></td>
+    <td><strong>Possible Match Methods</strong></td>
+  </tr>
+  <tr>
+    <td>1</td>
+    <td>An ID which is tied to a specific web browser or device (cookie-based, probabilistic, or other).</td>
+    <td>Any match method that pertains to one and only one browser and/or device, regardless of the method used to do the match. Any single mm value of 1, 2, 5, 6 could be applicable</td>  
+  </tr>
+  <tr>
+    <td>2</td>
+    <td>In-app impressions, which will typically contain a type of device ID (or rather, the privacy-compliant versions of device IDs).</td>
+    <td>Match method that pertains to a single application on a single device, regardless of how the match was done. Applicable mm values could be either 1, 5 or 6</td>  
+  </tr>
+  <tr>
+    <td>3</td>
+    <td>A person-based ID, i.e., that is the same across devices.</td>
+    <td>Match method that pertains to multiple devices, regardless of how the match was done. Applicable mm values could be either 3, 4</td> 
+  </tr>
+</table>
+
+### 7.12.4 - json Examples
+
+#### 7.12.4.1 User and Device IDs
+
+##### Example 1. DSP cookie derived from real-time cookie sync, subsequently observable
+```
+{
+    "user": {
+        "buyeruid": "fac13741-0648-436a-88cf-aceafdf45c9a",
+        "eids": [
+            {
+                "uids": [
+                    {
+                        "atype": "1"
+                    }
+                ]
+            }
+        ]
+    }
+}
+```
+
+##### Example 2. Resettable advertising ID generated by operating system
+```
+{
+    "device": {
+        "ifa": "df3a42f4-a388-43e9-a691-b0d308c963b4",
+        "eids": [
+            {
+                "uids": [
+                    {
+                        "atype": "2"
+                    }
+                ]
+            }
+        ]
+    }
+}
+```
+
+#### 7.12.4.2 Derrived IDs for a Single Property
+There exist scenarios in which a party may use an alternate approach to infer a DSP cookie user ID or IFA that is believed to apply or be related to the user, but cannot be directly observed. These scenarios should be conveyed using the below mechanism.
+
+##### Example 3: Request from Pub → SSP, user ID matched on hashed email, cookie not observable by SSP in the customary way
+
+```
+{
+    "user": {
+        "eids": [
+            {
+            "inserter": "pub.com",
+            "matcher": "graphvendor.com",
+            "source": "sspdomain.com”,
+            "mm": "3”,
+                "uids": [
+                    {
+                        "id": "fac13741-0648-436a-88cf-aceafdf45c9a",
+			"atype": "1"
```

Comment:
shouldn't this be "atype":3, since it's a hashed email, which is the same across devices?